js/apps/admin-ui/src/clients/authorization/policy/PolicyDetails.tsx

```diff
@@ -164,9 +172,12 @@ export default function PolicyDetails() {
     return <KeycloakSpinner />;
   }
 
-  const ComponentType = isValidComponentType(policyType)
-    ? COMPONENTS[policyType]
-    : COMPONENTS["js"];
+  let ComponentType = COMPONENTS["js-disabled"];
+  if (!IsDisabled) {
+    ComponentType = isValidComponentType(policyType)
+      ? COMPONENTS[policyType]
+      : COMPONENTS["js"];
+  }
```

Comment:
This seems to be getting a bit messy, perhaps we can extract this into a separate function so we can use an eager return pattern? For example:

```ts
function getComponentType(policyType: string, isDisabled: boolean) {
  if (isDisabled) {
    return COMPONENTS["js-disabled"];
  }

  return isValidComponentType(policyType)
    ? COMPONENTS[policyType]
    : COMPONENTS["js"];
}
```